src/components/Button/Button.spec.js

```diff
@@ -0,0 +1,82 @@
+import React from 'react'
+import { render, fireEvent } from '@testing-library/react'
+
+import { renderWithTheme, theme } from '../../../test/utils'
+import { blockSizes } from '../common/system'
+
+import Button from './Button'
+
+const defaultProps = {
+  children: 'click me'
+}
+
+describe('<Button />', () => {
+  it('should render button', () => {
+    const { getByRole } = render(<Button {...defaultProps} />)
+    const button = getByRole('button')
+
+    expect(button).toBeInTheDocument()
+    expect(button.tagName).toBe('BUTTON')
+  })
+
+  it('should handle different types', () => {
+    const { getByRole } = render(<Button {...defaultProps} type='submit' />)
+    const button = getByRole('button')
+
+    expect(button).toHaveAttribute('type', 'submit')
+  })
+
+  it('should handle click properly', () => {
+    const onButtonClick = jest.fn()
+    const { getByRole } = render(<Button {...defaultProps} onClick={onButtonClick} />)
+    const button = getByRole('button')
+
+    fireEvent.click(button)
+
+    expect(onButtonClick).toHaveBeenCalled()
+  })
+
+  it('should render disabled', () => {
```

Comment:
Here we should also test other variants of `<Button />` . That's because `variant="menu"` and `variant="flat"` also have the `disabled` option. Below links to the Storybook where you can see both variants in disabled and not disabled state:
[`variant="menu"`](https://arturbien.github.io/React95/?path=/story/button--menu-button)
[`variant="flat"`](https://arturbien.github.io/React95/?path=/story/button--flat-button)

Reply:
cool, added more test cases for different variants :)